packages/react/src/primitives/PhoneNumberField/utils.ts

```diff
@@ -0,0 +1,46 @@
+export const formatPhoneNumber = (
+  dialCode: string,
+  phoneNumberString: string
+): string => {
+  const cleaned = ('' + phoneNumberString).replace(/\D/g, '');
+  let formatted: string;
+  switch (dialCode) {
+    case '+1':
+      formatted = cleaned.replace(/(\d{3})(\d{3})(\d{4})/, '($1) $2-$3');
+      break;
```

Comment:
Where are these regex patterns coming from and what if A) I don't want this formatting B) I want a format of `'$1-$2-$3'` or `'$1.$2.$3'`?

Reply:
Are you suggesting we should have different formats for the same dial code?

Reply:
Suggesting that the current approach is inflexible for consumers and will be difficult to maintain from our side. What can we do that will allow the scenarios outlined in A and B above to be handled?

Reply:
Scenario A: we can add a new prop `isFormatted` that could be off by default, so users can control if they want to format the phone number or not.

Scenario B: I think we should not have different formats for let's say the same dial code, because then the regex logic would be very complex, unless you're ok with using a third party library like `libphonenumber` which is a library that parses, formats, stores and validates international phone numbers.

Let me know if you have other suggestions.